dvc/repo/experiments/queue/base.py

```diff
@@ -129,12 +129,17 @@ def remove(self, revs: Collection[str]) -> List[str]:
         to_remove = {}
         removed: List[str] = []
         for stash_rev, stash_entry in self.stash.stash_revs.items():
-            if stash_rev in revs:
-                to_remove[stash_rev] = stash_entry
-                removed.append(stash_rev)
-            elif stash_entry.name in revs:
-                to_remove[stash_rev] = stash_entry
-                removed.append(stash_entry.name)
+            for rev in revs:
+                if len(rev) < 4:
+                    continue
```

Comment:
I think we should be using `match_queue_entry_by_name` here. Once we have the matching `QueueEntry`'s, we can easily get the corresponding `StashEntry`'s to pass into `_remove_revs` with something like:
```py
queue_entries = match_queue_entry_by_name(revs, ...)
to_remove = {
    entry.stash_rev: self.stash.stash_revs[entry.stash_rev]
    for entry in queue_entries.values()
}
self._remove_revs(to_remove)
```